\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% title info
\title{Mix-ups and mixtures \\
  in microbiome data in DO mice}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics \\ Univ.\ Wisconsin{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[2pt]
\scriptsize {\lolit Slides:} \href{https://bit.ly/2019CTC}{\tt \scriptsize
  \color{foreground} bit.ly/2019CTC}
}


\begin{document}

% title slide
{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

  \vfill \hfill \includegraphics[height=6mm]{Figs/cc-zero.png} \vspace*{-3mm}

  \note{These are slides for a talk I'm going to give at the Complex
    Trait Community meeting in San Diego on 10 June 2019.

    Source: {\tt https://github.com/kbroman/Talk\_CTC2019} \\
    Slides: {\tt https://bit.ly/2019CTC}
    Slides with notes: {\tt https://bit.ly/2019CTC\_notes}
}
} }




\begin{frame}[c]{Microbiome genetics data}

\figw{Figs/overview.pdf}{1.0}

\note{
}

\end{frame}





\begin{frame}[c]{Multi-parent advanced intercross population}

\figw{Figs/hs.pdf}{1.0}

\note{
}

\end{frame}



\begin{frame}[c]{Genome of a diversity outbred mouse}

\figw{Figs/do_genome.pdf}{1.0}

\note{
}

\end{frame}


\begin{frame}[c]{Genotype reconstruction}

\only<1|handout 0>{\figw{Figs/geno_reconstruct.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/geno_reconstruct_B.pdf}{1.0}}
\only<3|handout 0>{\figw{Figs/geno_reconstruct_C.pdf}{1.0}}
\only<4>{\figw{Figs/geno_reconstruct_D.pdf}{1.0}}

\note{
}

\end{frame}




\begin{frame}[c]{Mapped reads}

\only<1|handout 0>{\figw{Figs/mapped_reads.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/mapped_reads_B.pdf}{1.0}}
\only<3>{\figw{Figs/mapped_reads_C.pdf}{1.0}}


\note{
}

\end{frame}


\begin{frame}[c]{Genomic DNA vs microbiome reads}


\onslide<2->{\bigskip {\hilit genomic DO-381 vs microbiome DO-381}}


\only<1|handout 0>{
  \begin{center}
      \renewcommand{\arraystretch}{2}
    \begin{tabular}{c|r|r|}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{microbiome DNA}} \\
    \multicolumn{1}{c}{\textbf{genomic DNA}} & \multicolumn{1}{c}{\textbf{A}}
           & \multicolumn{1}{c}{\textbf{B}} \\ \cline{2-3}
    \textbf{AA} &        &             \\ \cline{2-3}
    \textbf{AB} &        &             \\ \cline{2-3}
    \textbf{BB} &        &             \\ \cline{2-3}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\color{white} percent mismatch} \\
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\hspace*{23mm}} &
            \multicolumn{1}{c}{\hspace*{23mm}}
  \end{tabular}
  \end{center}
}


\only<2->{
  \begin{center}
      \renewcommand{\arraystretch}{2}
    \begin{tabular}{c|r|r|}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{microbiome DNA}} \\
    \multicolumn{1}{c}{\textbf{genomic DNA}} & \multicolumn{1}{c}{\textbf{A}}
           & \multicolumn{1}{c}{\textbf{B}} \\ \cline{2-3}
    \textbf{AA} & 2,762,341 &   7,303 \\ \cline{2-3}
    \textbf{AB} &   606,312 & 578,017 \\ \cline{2-3}
    \textbf{BB} &     2,128 & 375,559 \\ \cline{2-3}
     \multicolumn{1}{c}{} & \multicolumn{2}{c}{
       \only<3>{\hilit percent mismatch = 0.3\%}} \\
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\hspace*{23mm}} &
            \multicolumn{1}{c}{\hspace*{23mm}}
  \end{tabular}
  \end{center}
}


\note{
}

\end{frame}


\begin{frame}[c]{Genomic DNA vs microbiome reads}


\bigskip {\hilit genomic DO-360 vs microbiome DO-360}


  \begin{center}
      \renewcommand{\arraystretch}{2}
    \begin{tabular}{c|r|r|}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{microbiome DNA}} \\
    \multicolumn{1}{c}{\textbf{genomic DNA}} & \multicolumn{1}{c}{\textbf{A}}
           & \multicolumn{1}{c}{\textbf{B}} \\ \cline{2-3}
    \textbf{AA} & 8,863,572 & 1,520,169 \\ \cline{2-3}
    \textbf{AB} & 2,870,063 & 1,075,126 \\ \cline{2-3}
    \textbf{BB} &   671,722 &  536,010  \\ \cline{2-3}
     \multicolumn{1}{c}{} & \multicolumn{2}{c}{\hilit percent mismatch = 19\%} \\
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\hspace*{23mm}} &
            \multicolumn{1}{c}{\hspace*{23mm}}
  \end{tabular}
  \end{center}


\note{
}

\end{frame}


\begin{frame}[c]{Genomic DNA vs microbiome reads}


\bigskip {\hilit genomic DO-360 vs microbiome DO-370}

  \begin{center}
      \renewcommand{\arraystretch}{2}
    \begin{tabular}{c|r|r|}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{microbiome DNA}} \\
    \multicolumn{1}{c}{\textbf{genomic DNA}} & \multicolumn{1}{c}{\textbf{A}}
           & \multicolumn{1}{c}{\textbf{B}} \\ \cline{2-3}
    \textbf{AA} & 3,409,773 &   11,311 \\ \cline{2-3}
    \textbf{AB} &   659,986 &  615,628 \\ \cline{2-3}
    \textbf{BB} &     3,375 &  383,222 \\ \cline{2-3}
     \multicolumn{1}{c}{} & \multicolumn{2}{c}{\hilit percent mismatch = 0.4\%} \\
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\hspace*{23mm}} &
            \multicolumn{1}{c}{\hspace*{23mm}}
  \end{tabular}
  \end{center}


\note{
}

\end{frame}






\begin{frame}[c]{Distance matrix}

\figw{Figs/dist_matrix.pdf}{1.0}

\note{
}

\end{frame}



\begin{frame}[c]{Minimum vs. self distance}

\figw{Figs/min_v_self.pdf}{1.0}

\note{
}

\end{frame}



\begin{frame}[c]{Detailed distances}

\figw{Figs/detailed_mixups.pdf}{1.0}

\note{
}

\end{frame}



\begin{frame}[c]{Genotype pairs vs microbiome reads}

\bbi
 \item table with pairs of genotypes vs microbiome reads
 \item then show table for a clean sample
 \item then show probability model
\ei

\note{
}

\end{frame}



\begin{frame}[c]{log likelihood vs contaminant proproportion}

\bbi
 \item scatterplot of log likelihood vs contaminant proportion
\ei

\note{
}

\end{frame}



\begin{frame}[c]{Is everything contaminated?}

\bbi
 \item expand the lower-left corner
\ei

\note{
}

\end{frame}




\begin{frame}[c]{Summary}

\bbi
 \item everyone likes a good summary
\ei

\note{
}

\end{frame}




\begin{frame}[c]{Acknowledgments}

\bbi
 \item Lindsay Traeger
 \item Alexandra Lobo
 \item Federico Rey
 \item Alan Attie, Mark Keller, Gary Churchill, Brian Yandell
\ei

\note{
}

\end{frame}






\begin{frame}[c]{}

\large

Slides: \href{https://bit.ly/2019CTC}{\tt bit.ly/2019CTC} \quad
\includegraphics[height=5mm]{Figs/cc-zero.png}

\vspace{8mm}

bioRxiv manuscript: \href{https://doi.org/10.1101/529040}{{\tt doi.org/10.1101/529040}}

\vspace{8mm}

\href{https://kbroman.org}{\tt kbroman.org}

\vspace{8mm}

\href{https://github.com/kbroman}{\tt github.com/kbroman}

\vspace{8mm}

\href{https://twitter.com/kwbroman}{\tt @kwbroman}


\note{
  Here's where you can find me, as well as the slides for this talk.

  Also note that there is a bioRxiv manuscript describing the details
  of this work.
}
\end{frame}




\end{document}
