\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% title info
\title{Sample mix-ups and mixtures \\
  in microbiome data in DO mice}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics \\ Univ.\ Wisconsin{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[2pt]
\scriptsize {\lolit Slides:} \href{https://bit.ly/2019CTC}{\tt \scriptsize
  \color{foreground} bit.ly/2019CTC}
}


\begin{document}

% title slide
{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

  \vfill \hfill \includegraphics[height=6mm]{Figs/cc-zero.png} \vspace*{-3mm}

  \note{These are slides for a talk I'm going to give at the Complex
    Trait Community meeting ({\tt http://ratgenes.org/ctc2019}) in San
    Diego on 10 June 2019.

    Source: {\tt https://github.com/kbroman/Talk\_CTC2019} \\
    Slides: {\tt https://bit.ly/2019CTC} \\
    Slides with notes: {\tt https://bit.ly/2019CTC\_notes}
}
} }




\begin{frame}[c]{Microbiome genetics data}

\figw{Figs/overview.pdf}{1.0}

\note{
  As part of a larger project seeking to understand the genetics of
  diabetes and obesity, we're looking at genetic effects on the
  microbiome. We have a set of about 500 diversity outbred mice, with
  SNP genotypes from GigaMUGA arrays. For 300 mice, we have shotgun
  sequencing data on DNA extracted from mouse poop.

  The goal of the microbiome sequencing was to characterize the
  bacteria in the gut of the mice, but the data also include reads
  derived from the mouse host. This offers an opportunity to
  check for sample duplicates between the microbiome samples and the
  genomic DNA samples.

  We want to relate the genotypes at SNPs across the genome to the
  alleles observed in the microbiome sequence reads.
}

\end{frame}





\begin{frame}[c]{Multi-parent advanced intercross population}

\figw{Figs/hs.pdf}{1.0}

\note{
Diversity outbred mice, like heterogeneous stock, are an example of a
multi-parent advanced intercross population. Eight founder strains
were inter-bred for many generations, maintaining as large a
population as is feasible at each generation and avoiding crosses
between siblings, to avoid inbreeding and genetic drift.

The resulting population is a heterogeneous mixture of the initial
eight strains, with the chromosomes broken down into small pieces.
}

\end{frame}



\begin{frame}[c]{Genome of a diversity outbred mouse}

\figw{Figs/do_genome.pdf}{1.0}

\note{
This is an example of the genome of one DO mouse. At any one position,
they have one of 36 possible genotypes. The white patches are regions
where genotype is uncertain.
}

\end{frame}


\begin{frame}[c]{Genotype reconstruction}

\only<1|handout 0>{\figw{Figs/geno_reconstruct.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/geno_reconstruct_B.pdf}{1.0}}
\only<3|handout 0>{\figw{Figs/geno_reconstruct_C.pdf}{1.0}}
\only<4>{\figw{Figs/geno_reconstruct_D.pdf}{1.0}}

\note{
Our first step is to reconstruct the founder genotypes along the
genome of each DO mouse, using SNP data on DO mice and the eight
founder strains. We use a hidden Markov model for this purpose, which
allows for the presence of genotyping errors.

The white, gray, and black circles indicate SNP genotypes AA, AB, and
BB, with A being the major allele and B being the minor allele (based
on frequency in the eight founder strains.)

The eight founder strains have been sequenced, and so we know their
genotype at about 40 million polymorphisms in the genome. We can use
those genotypes plus the DO genotype reconstructions to infer the DO
mouse genotypes at all SNPs.

This is important because we want to compare sequence reads to the SNP
genotypes, and while many sequence reads will overlap a SNP, few will
overlap one of the SNPs on the GigaMUGA array, for which we have
direct DO genotype data.
}



\end{frame}




\begin{frame}[c]{Mapped reads}

\only<1|handout 0>{\figw{Figs/mapped_reads.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/mapped_reads_B.pdf}{1.0}}
\only<3>{\figw{Figs/mapped_reads_C.pdf}{1.0}}


\note{
The second part of the process is to map sequence reads to the mouse
genome. We then look at the SNPs in a region, identify which reads
overlap a SNP, and the count the observed alleles at the different reads.

If a DO mouse really corresponds to that microbiome sample, all reads
should be A whereever the mouse has genotype AA, and B where the mouse
is BB. At SNPs where the mouse is AB, half the reads should be A and
half should be B.

There will be some errors in the sequence reads, such as the black dot
in the interval 45.002--45.004, corresponding to an allele B in a read
at a SNP where the mouse is AA.
}

\end{frame}


\begin{frame}[c]{Genomic DNA vs microbiome reads}


\onslide<2->{\bigskip {\hilit genomic DO-381 vs microbiome DO-381}}


\only<1|handout 0>{
  \begin{center}
      \renewcommand{\arraystretch}{2}
    \begin{tabular}{c|r|r|}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{microbiome DNA}} \\
    \multicolumn{1}{c}{\textbf{genomic DNA}} & \multicolumn{1}{c}{\textbf{A}}
           & \multicolumn{1}{c}{\textbf{B}} \\ \cline{2-3}
    \textbf{AA} &        &             \\ \cline{2-3}
    \textbf{AB} &        &             \\ \cline{2-3}
    \textbf{BB} &        &             \\ \cline{2-3}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\color{white} percent mismatch} \\
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\hspace*{23mm}} &
            \multicolumn{1}{c}{\hspace*{23mm}}
  \end{tabular}
  \end{center}
}


\only<2->{
  \begin{center}
      \renewcommand{\arraystretch}{2}
    \begin{tabular}{c|r|r|}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{microbiome DNA}} \\
    \multicolumn{1}{c}{\textbf{genomic DNA}} & \multicolumn{1}{c}{\textbf{A}}
           & \multicolumn{1}{c}{\textbf{B}} \\ \cline{2-3}
    \textbf{AA} & 2,762,341 &   7,303 \\ \cline{2-3}
    \textbf{AB} &   606,312 & 578,017 \\ \cline{2-3}
    \textbf{BB} &     2,128 & 375,559 \\ \cline{2-3}
     \multicolumn{1}{c}{} & \multicolumn{2}{c}{
       \only<3>{\hilit percent mismatch = 0.3\%}} \\
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\hspace*{23mm}} &
            \multicolumn{1}{c}{\hspace*{23mm}}
  \end{tabular}
  \end{center}
}


\note{
  Our main strategy is to split all SNPs according to their genotype
  in a genomic DNA sample, with AA meaning homozygous for the major
  allele (among the eight founding strains). For a microbiome sample,
  we count the number of overlapping reads with the A vs. the B allele.

  If the microbiome sample really comes from that mouse, the reads
  should be largely A at the AA SNPs, and largely B at the BB SNPs,
  and about 50:50 A vs B at the heterozygous SNPs.

  In this particular case, the percent mismatches in the homozygous
  SNPs is 0.3\%. You could take that as an estimate of the sequencing
  error rate, assuming there's no mix-up here, and that the genomic
  DNA genotypes are correct.
}

\end{frame}


\begin{frame}[c]{Genomic DNA vs microbiome reads}


\bigskip {\hilit genomic DO-360 vs microbiome DO-360}


  \begin{center}
      \renewcommand{\arraystretch}{2}
    \begin{tabular}{c|r|r|}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{microbiome DNA}} \\
    \multicolumn{1}{c}{\textbf{genomic DNA}} & \multicolumn{1}{c}{\textbf{A}}
           & \multicolumn{1}{c}{\textbf{B}} \\ \cline{2-3}
    \textbf{AA} & 8,863,572 & 1,520,169 \\ \cline{2-3}
    \textbf{AB} & 2,870,063 & 1,075,126 \\ \cline{2-3}
    \textbf{BB} &   671,722 &  536,010  \\ \cline{2-3}
     \multicolumn{1}{c}{} & \multicolumn{2}{c}{\hilit percent mismatch = 19\%} \\
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\hspace*{23mm}} &
            \multicolumn{1}{c}{\hspace*{23mm}}
  \end{tabular}
  \end{center}


\note{
   This case, of DO-360 genomic DNA vs DO-360 microbiome, shows a
   clearly different pattern: 19\% mismatch in the homozygous SNPs,
   and the heterozygous SNPs show A:B like 3:1.
}

\end{frame}


\begin{frame}[c]{Genomic DNA vs microbiome reads}


\bigskip {\hilit genomic DO-370 vs microbiome DO-360}

  \begin{center}
      \renewcommand{\arraystretch}{2}
    \begin{tabular}{c|r|r|}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{microbiome DNA}} \\
    \multicolumn{1}{c}{\textbf{genomic DNA}} & \multicolumn{1}{c}{\textbf{A}}
           & \multicolumn{1}{c}{\textbf{B}} \\ \cline{2-3}
    \textbf{AA} & 10,324,265 &    23,256 \\ \cline{2-3}
    \textbf{AB} &  2,083,947 & 1,986,380 \\ \cline{2-3}
    \textbf{BB} &      5,347 & 1,117,994 \\ \cline{2-3}
     \multicolumn{1}{c}{} & \multicolumn{2}{c}{\hilit percent mismatch = 0.2\%} \\
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\hspace*{23mm}} &
            \multicolumn{1}{c}{\hspace*{23mm}}
  \end{tabular}
  \end{center}


\note{
   If we compare the DO-360 microbiome to genomic DNA for DO-370,
   however, we see nice concordance.

   While the heterozygous category does contain information, we're
   going to just focus on homozygous SNPs and use the percent mismatch
   as a measure of distance.
}

\end{frame}






\begin{frame}[c]{Distance matrix}

\figw{Figs/dist_matrix.pdf}{1.0}

\note{
  This shows the distances between microbiome samples (on y-axis) and
  genomic DNA samples (on x-axis), with black indicating the samples
  are similar, and white indicating they are different.

  There were a total of 500 DO mice, in five batches of 100 mice each.
  The microbiome study included the first, second, and fourth batches.
  The black diagonal line indicates that most of the samples are
  correct; but if you look closely you can see some problems.
}

\end{frame}



\begin{frame}[c]{Minimum vs. self distance}

\figw{Figs/min_v_self.pdf}{1.0}

\note{
  Here, we plot the minimum distance for each microbiome sample (the
  minimum value in each row in the distance matrix) vs.\ the self-self
  distance.

  Samples on the diagonal (in light blue) look to be
  correctly labeled: they are similar to the corresponding genomic DNA
  sample and no other sample is closer.

  Samples in the lower-right corner (purple) are mix-ups. They are very
  different from the correspondingly labeled sample, but there is
  another sample that is quite close.

  Samples in green turn out to be mixtures (more on this below).

  Samples in pink (upper right) have bad genomic DNA samples. Samples
  in light purple have low read counts.
}

\end{frame}



\begin{frame}[c]{Selected samples}

\figw{Figs/detailed_mixups.pdf}{1.0}

\note{
   These are the results for selected samples, of the distance between
   a microbiome sample and all 500 genomic DNA samples (the values along a
   row of the distance matrix).

   The dark pink dot is for the genomic DNA sample with the sample
   label. The light pink dots are a set of low-quality
   genomic DNA samples.

   The left panels are a clear mix-up between DO-53 and DO-54.
   The center panels are a clear mix-up between DO-360 and DO-370.

   The upper-right panel shows that DO-361 is well behaved.

   The lower-right panel is for the microbiome sample DO-362. It is
   most similar to genomic DNA sample DO-361, but it's not too
   similar, and the second-most similar genomic DNA sample is the
   one with the same label as the microbiome sample, DO-362.
   This suggested that the microbiome sample is perhaps a mixture
   between DO-361 and DO-362.
}

\end{frame}



\begin{frame}{Genotype pair vs microbiome reads}


\bigskip {\hilit genomic DO-362 and DO-361 vs microbiome DO-362}

\bigskip \bigskip

{\hspace*{-9mm}
  \footnotesize
  \renewcommand{\arraystretch}{2}
  \setlength{\tabcolsep}{1.5mm}
\only<1|handout 0>{ \begin{tabular}{c|r|r|cc|r|r|cc|r|r|}
        \multicolumn{1}{c}{\textbf{DO-362:}} & \multicolumn{2}{c}{\textbf{AA}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{AB}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{BB}} \\

        \multicolumn{1}{c}{\textbf{DO-361}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-361}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-361}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} \\
        \cline{2-3}\cline{6-7}\cline{10-11}
      \textbf{AA} & 1,306,406 &   3,688 &&
      \textbf{AA} &   395,562 &  75,326 &&
      \textbf{AA} &    41,593 &  19,813 \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{AB} & 339,397 & 162,432 &&
      \textbf{AB} & 135,825 & 128,895 &&
      \textbf{AB} &  25,798 &  49,487 \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{BB} & 33,661 & 63,318 &&
      \textbf{BB} & 14,836 & 71,817 &&
      \textbf{BB} &    529 & 93,162 \\
        \cline{2-3}\cline{6-7}\cline{10-11}

    \multicolumn{1}{c}{} & \multicolumn{1}{c}{} & \multicolumn{1}{c}{} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{} & \multicolumn{1}{c}{} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{} & \multicolumn{1}{c}{}
  \end{tabular} }
\only<2>{  \begin{tabular}{c|r|r|cc|r|r|cc|r|r|}
        \multicolumn{1}{c}{\textbf{DO-362:}} & \multicolumn{2}{c}{\textbf{AA}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{AB}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{BB}} \\

        \multicolumn{1}{c}{\textbf{DO-361}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-361}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-361}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} \\
        \cline{2-3}\cline{6-7}\cline{10-11}
      \textbf{AA} & 99.7\% &  0.3\% &&
      \textbf{AA} & 84.0\% & 16.0\% &&
      \textbf{AA} & 67.7\% & 32.3\% \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{AB} & 67.6\%   & 32.4\%  &&
      \textbf{AB} & 51.3\%   & 48.7\%  &&
      \textbf{AB} & 34.3\%   & 65.7\%  \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{BB} & 34.7\% & 65.3\% &&
      \textbf{BB} & 17.1\% & 82.9\% &&
      \textbf{BB} &  0.6\% & 99.4\% \\
        \cline{2-3}\cline{6-7}\cline{10-11}

    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\color{white} 1,306,406}
                & \multicolumn{1}{c}{\color{white} 75,326} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\color{white} 339,397}
                & \multicolumn{1}{c}{\color{white} 162,432} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\color{white} 33,661}
                & \multicolumn{1}{c}{\color{white} 93,162}
  \end{tabular} }
}


\note{
    To assess whether the microbiome sample DO-362 is in fact a
    mixture, we can divide SNPs into nine groups according to the
    genotypes of DO-362 and DO-361, and then count the number of A and
    B reads in each group.

    It should be that the frequency of A vs. B reads in the microbiome
    sample DO-362 should only depend on the genotype of DO-362 and not
    on the genotype of DO-361. But as shown on this slide, there is a
    very strong dependence on the genotype of DO-361.

    For example, when DO-362 is AA, there should be a small proportion
    of B reads, irrespective of what the genotype of DO-361 is. But
    when DO-361 is BB, there are about 65\% B reads. It seems clear
    that the DO-362 microbiome sample is contaminated with DNA from
    DO-361, and in fact it could be about 65\% from DO-361.
}

\end{frame}



\begin{frame}{Genotype pair vs microbiome reads}


\bigskip {\hilit genomic DO-361 and DO-362 vs microbiome DO-361}

\bigskip \bigskip

{\hspace*{-9mm}
  \footnotesize
  \renewcommand{\arraystretch}{2}
  \setlength{\tabcolsep}{1.5mm}
\only<1|handout 0>{  \begin{tabular}{c|r|r|cc|r|r|cc|r|r|}
        \multicolumn{1}{c}{\textbf{DO-361:}} & \multicolumn{2}{c}{\textbf{AA}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{AB}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{BB}} \\

        \multicolumn{1}{c}{\textbf{DO-362}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-362}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-362}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} \\
        \cline{2-3}\cline{6-7}\cline{10-11}
      \textbf{AA} & 2,162,059 &   6,503 &&
      \textbf{AA} &   428,229 & 405,190 &&
      \textbf{AA} &     1,623 & 155,276 \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{AB} & 788,011 &   2,806 &&
      \textbf{AB} & 223,680 & 215,833 &&
      \textbf{AB} &   1,126 & 140,739 \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{BB} & 104,511 &     511 &&
      \textbf{BB} &  64,082 &  62,658 &&
      \textbf{BB} &   1,103 & 151,334 \\
        \cline{2-3}\cline{6-7}\cline{10-11}

    \multicolumn{1}{c}{} & \multicolumn{1}{c}{} & \multicolumn{1}{c}{} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{} & \multicolumn{1}{c}{} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{} & \multicolumn{1}{c}{}
  \end{tabular} }
\only<2>{  \begin{tabular}{c|r|r|cc|r|r|cc|r|r|}
        \multicolumn{1}{c}{\textbf{DO-361:}} & \multicolumn{2}{c}{\textbf{AA}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{AB}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{BB}} \\

        \multicolumn{1}{c}{\textbf{DO-362}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-362}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-362}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} \\
        \cline{2-3}\cline{6-7}\cline{10-11}
      \textbf{AA} & 99.7\% &  0.3\% &&
      \textbf{AA} & 51.4\% & 48.6\%  &&
      \textbf{AA} &  1.0\% & 99.0\% \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{AB} & 99.6\% &  0.4\% &&
      \textbf{AB} & 50.9\% & 49.1\%  &&
      \textbf{AB} &  0.8\% & 99.2\%  \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{BB} & 99.5\% &  0.5\% &&
      \textbf{BB} & 50.6\% & 49.4\% &&
      \textbf{BB} &  0.7\% & 99.3\% \\
        \cline{2-3}\cline{6-7}\cline{10-11}

    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\color{white} 2,162,059}
                & \multicolumn{1}{c}{\color{white} 6,503} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\color{white} 428,229}
                & \multicolumn{1}{c}{\color{white} 405,190} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\color{white} 1,623}
                & \multicolumn{1}{c}{\color{white} 155,276}
  \end{tabular} }
}

\note{
  Here's a similar table for the DO-361 microbiome sample. This is
  what a clean microbiome sample should look like. But note that there is
  still some small association with the genotype of the DO-362 sample.
}

\end{frame}



\begin{frame}[c]{log likelihood vs contaminant proportion}

\figw{Figs/lrt_v_contam.pdf}{1.0}

\note{
  To evaluate the potential for mixtures more formally, we fit a model
  where we assume that a microbiome sample is a mixture of its own DNA
  and some proportion $p$ of DNA from one other sample, and that the
  number of A and B microbiome reads are binomial counts with
  sequencing error rate $e$.

  This figure shows the estimated proportion contaminant (on the
  x-axis) and the likelihood ratio statistic for the test of $p=0$ (on
  the y-axis). On the far right (in purple) are the sample mix-ups. In
  green are a variety of mixtures, most with $>$ 50\% coming from the
  contaminant.

  Note that the likelihood ratio statistics are huge. We're looking
  for something like 10, and we're getting values $>$ 1 million.
}

\end{frame}




\begin{frame}[c]{Selected samples}

\figw{Figs/lrt_v_contam_detail.pdf}{1.0}

\note{
  Here are results for a selected set of microbiome samples. Each
  sample considers the microbiome sample as a mixture of the
  corresponding genomic DNA sample and one contaminant, and each point
  is one of the 499 possible contaminants.

  Each panel shows the likelihood ratio test statistic for the test of
  $p=0$ (on the y-axis) vs.\ the estimated contaminant proportion (on
  the x-axis). In all cases, a single contaminant stands out.

  The left panels are for one of the mix-ups. The other four panels
  are for different apparent mixtures.
}

\end{frame}



\begin{frame}[c]{Is everything contaminated?}

\figw{Figs/lrt_v_contam_expand.pdf}{1.0}

\note{
  If we expand the lower-left corner of the overview figure, we find
  lots of samples that have strong evidence for being mixtures. Note
  that the LRT statistics on the y-axis have been divided by 10,000.

  It's hard to draw a line on what is a real mixture and what is just
  noise, and we probably don't need to worry about cases that are $<$
  5\% contaminant; for studying the genetics of the microbiome
  composition, that could be viewed as acceptable phenotypic noise.
}

\end{frame}




\begin{frame}[c]{Summary}

\bbi
 \item Microbiome shotgun reads include reads from the host
 \item With such data, sample mix-ups can be identified
 \item Simple method:
   \bi
 \item Impute genotype at all SNPs
 \item Count alleles in reads overlapping SNPs
 \item Focus on homozygous SNPs and calculate percent discordant reads
   \ei
 \item We also saw strong evidence for many samples being mixtures
\ei

\note{
It is always important to provide a summary.
}

\end{frame}




\begin{frame}[c]{Acknowledgments}

\bbi
 \item Lindsay Traeger
 \item Alexandra Lobo
 \item Federico Rey
 \item Alan Attie, Mark Keller, Gary Churchill, Brian Yandell
\ei

\note{
  Lindsay had the idea to look for these mix-ups. (She was a postdoc
  with Federico Rey in Microbiology at UW-Madison.) Alexandra did most
  of the work. (She was a summer student with me and now is a graduate
  student in the Biomedical Data Science PhD program at UW-Madison.

  This is part of a larger project looking at the genetics of
  diabetes, obesity, and related traits.
}

\end{frame}






\begin{frame}[c]{}

\large

Slides: \href{https://bit.ly/2019CTC}{\tt bit.ly/2019CTC}
\quad \quad
\includegraphics[height=5mm]{Figs/cc-zero.png} \\
\hspace{15.7mm} \href{https://bit.ly/2019CTC_notes}{\tt bit.ly/2019CTC\_notes}

\vspace{8mm}

bioRxiv manuscript: \href{https://doi.org/10.1101/529040}{{\tt doi.org/10.1101/529040}}

\vspace{8mm}

\href{https://kbroman.org}{\tt kbroman.org}

\vspace{8mm}

\href{https://github.com/kbroman}{\tt github.com/kbroman}

\vspace{8mm}

\href{https://twitter.com/kwbroman}{\tt @kwbroman}


\note{
  Here's where you can find me, as well as the slides for this talk.

  Also note that there is a bioRxiv manuscript describing the details
  of this work.
}
\end{frame}




\end{document}
