\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}

\input{header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% title info
\title{Sample mix-ups and mixtures \\
  in microbiome data in DO mice}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics \\ Univ.\ Wisconsin{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[2pt]
\scriptsize {\lolit Slides:} \href{https://bit.ly/2019CTC}{\tt \scriptsize
  \color{foreground} bit.ly/2019CTC}
}


\begin{document}

% title slide
{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

  \vfill \hfill \includegraphics[height=6mm]{Figs/cc-zero.png} \vspace*{-3mm}

  \note{These are slides for a talk I'm going to give at the Complex
    Trait Community meeting ({\tt http://ratgenes.org/ctc2019}) in San
    Diego on 10 June 2019.

    Source: {\tt https://github.com/kbroman/Talk\_CTC2019} \\
    Slides: {\tt https://bit.ly/2019CTC} \\
    Slides with notes: {\tt https://bit.ly/2019CTC\_notes}
}
} }




\begin{frame}[c]{Microbiome genetics data}

\figw{Figs/overview.pdf}{1.0}

\note{
  As part of a larger project seeking to understand the genetics of
  diabetes and obesity, we're looking at genetic effects on the
  microbiome. We have a set of about 500 diversity outbred mice, with
  SNP genotypes from GigaMUGA arrays. For 300 mice, we have shotgun
  sequencing data on DNA extracted from mouse poop.

  The goal of the microbiome sequencing was to characterize the
  bacteria in the gut of the mice, but the data also include reads
  derived from the mouse host. This offers an opportunity to
  check for sample duplicates between the microbiome samples and the
  genomic DNA samples.
}

\end{frame}





\begin{frame}[c]{Multi-parent advanced intercross population}

\figw{Figs/hs.pdf}{1.0}

\note{
Diversity outbred mice, like heterogeneous stock, are an example of a
multi-parent advanced intercross population. Eight founder strains
were inter-bred for many generations, maintaining as large a
population as is feasible at each generation and avoiding crosses
between siblings, to avoid inbreeding and genetic drift.

The resulting population is a heterogeneous mixture of the initial
eight strains, with the chromosomes broken down into small pieces.
}

\end{frame}



\begin{frame}[c]{Genome of a diversity outbred mouse}

\figw{Figs/do_genome.pdf}{1.0}

\note{
This is an example of the genome of one DO mouse. At any one position,
they have one of 36 possible genotypes. The white patches are regions
where genotype is uncertain.
}

\end{frame}


\begin{frame}[c]{Genotype reconstruction}

\only<1|handout 0>{\figw{Figs/geno_reconstruct.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/geno_reconstruct_B.pdf}{1.0}}
\only<3|handout 0>{\figw{Figs/geno_reconstruct_C.pdf}{1.0}}
\only<4>{\figw{Figs/geno_reconstruct_D.pdf}{1.0}}

\note{
}

\end{frame}




\begin{frame}[c]{Mapped reads}

\only<1|handout 0>{\figw{Figs/mapped_reads.pdf}{1.0}}
\only<2|handout 0>{\figw{Figs/mapped_reads_B.pdf}{1.0}}
\only<3>{\figw{Figs/mapped_reads_C.pdf}{1.0}}


\note{
}

\end{frame}


\begin{frame}[c]{Genomic DNA vs microbiome reads}


\onslide<2->{\bigskip {\hilit genomic DO-381 vs microbiome DO-381}}


\only<1|handout 0>{
  \begin{center}
      \renewcommand{\arraystretch}{2}
    \begin{tabular}{c|r|r|}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{microbiome DNA}} \\
    \multicolumn{1}{c}{\textbf{genomic DNA}} & \multicolumn{1}{c}{\textbf{A}}
           & \multicolumn{1}{c}{\textbf{B}} \\ \cline{2-3}
    \textbf{AA} &        &             \\ \cline{2-3}
    \textbf{AB} &        &             \\ \cline{2-3}
    \textbf{BB} &        &             \\ \cline{2-3}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\color{white} percent mismatch} \\
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\hspace*{23mm}} &
            \multicolumn{1}{c}{\hspace*{23mm}}
  \end{tabular}
  \end{center}
}


\only<2->{
  \begin{center}
      \renewcommand{\arraystretch}{2}
    \begin{tabular}{c|r|r|}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{microbiome DNA}} \\
    \multicolumn{1}{c}{\textbf{genomic DNA}} & \multicolumn{1}{c}{\textbf{A}}
           & \multicolumn{1}{c}{\textbf{B}} \\ \cline{2-3}
    \textbf{AA} & 2,762,341 &   7,303 \\ \cline{2-3}
    \textbf{AB} &   606,312 & 578,017 \\ \cline{2-3}
    \textbf{BB} &     2,128 & 375,559 \\ \cline{2-3}
     \multicolumn{1}{c}{} & \multicolumn{2}{c}{
       \only<3>{\hilit percent mismatch = 0.3\%}} \\
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\hspace*{23mm}} &
            \multicolumn{1}{c}{\hspace*{23mm}}
  \end{tabular}
  \end{center}
}


\note{
}

\end{frame}


\begin{frame}[c]{Genomic DNA vs microbiome reads}


\bigskip {\hilit genomic DO-360 vs microbiome DO-360}


  \begin{center}
      \renewcommand{\arraystretch}{2}
    \begin{tabular}{c|r|r|}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{microbiome DNA}} \\
    \multicolumn{1}{c}{\textbf{genomic DNA}} & \multicolumn{1}{c}{\textbf{A}}
           & \multicolumn{1}{c}{\textbf{B}} \\ \cline{2-3}
    \textbf{AA} & 8,863,572 & 1,520,169 \\ \cline{2-3}
    \textbf{AB} & 2,870,063 & 1,075,126 \\ \cline{2-3}
    \textbf{BB} &   671,722 &  536,010  \\ \cline{2-3}
     \multicolumn{1}{c}{} & \multicolumn{2}{c}{\hilit percent mismatch = 19\%} \\
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\hspace*{23mm}} &
            \multicolumn{1}{c}{\hspace*{23mm}}
  \end{tabular}
  \end{center}


\note{
}

\end{frame}


\begin{frame}[c]{Genomic DNA vs microbiome reads}


\bigskip {\hilit genomic DO-360 vs microbiome DO-370}

  \begin{center}
      \renewcommand{\arraystretch}{2}
    \begin{tabular}{c|r|r|}
    \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{microbiome DNA}} \\
    \multicolumn{1}{c}{\textbf{genomic DNA}} & \multicolumn{1}{c}{\textbf{A}}
           & \multicolumn{1}{c}{\textbf{B}} \\ \cline{2-3}
    \textbf{AA} & 3,409,773 &   11,311 \\ \cline{2-3}
    \textbf{AB} &   659,986 &  615,628 \\ \cline{2-3}
    \textbf{BB} &     3,375 &  383,222 \\ \cline{2-3}
     \multicolumn{1}{c}{} & \multicolumn{2}{c}{\hilit percent mismatch = 0.4\%} \\
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\hspace*{23mm}} &
            \multicolumn{1}{c}{\hspace*{23mm}}
  \end{tabular}
  \end{center}


\note{
}

\end{frame}






\begin{frame}[c]{Distance matrix}

\figw{Figs/dist_matrix.pdf}{1.0}

\note{
}

\end{frame}



\begin{frame}[c]{Minimum vs. self distance}

\figw{Figs/min_v_self.pdf}{1.0}

\note{
}

\end{frame}



\begin{frame}[c]{Detailed distances}

\figw{Figs/detailed_mixups.pdf}{1.0}

\note{
}

\end{frame}



\begin{frame}{Genotype pair vs microbiome reads}


\bigskip {\hilit genomic DO-362 and DO-361 vs microbiome DO-362}

\bigskip \bigskip

{\hspace*{-9mm}
  \footnotesize
  \renewcommand{\arraystretch}{2}
  \setlength{\tabcolsep}{1.5mm}
\only<1>{ \begin{tabular}{c|r|r|cc|r|r|cc|r|r|}
        \multicolumn{1}{c}{\textbf{DO-362:}} & \multicolumn{2}{c}{\textbf{AA}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{AB}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{BB}} \\

        \multicolumn{1}{c}{\textbf{DO-361}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-361}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-361}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} \\
        \cline{2-3}\cline{6-7}\cline{10-11}
      \textbf{AA} & 1,306,406 &   3,688 &&
      \textbf{AA} &   339,397 & 162,432 &&
      \textbf{AA} &    33,661 &  63,318 \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{AB} & 395,562 &  75,326 &&
      \textbf{AB} & 135,825 & 128,895 &&
      \textbf{AB} &  14,836 &  71,817 \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{BB} & 41,593 & 19,813 &&
      \textbf{BB} & 25,798 & 49,487 &&
      \textbf{BB} &    529 & 93,162 \\
        \cline{2-3}\cline{6-7}\cline{10-11}

    \multicolumn{1}{c}{} & \multicolumn{1}{c}{} & \multicolumn{1}{c}{} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{} & \multicolumn{1}{c}{} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{} & \multicolumn{1}{c}{}
  \end{tabular} }
\only<2|handout 0>{  \begin{tabular}{c|r|r|cc|r|r|cc|r|r|}
        \multicolumn{1}{c}{\textbf{DO-362:}} & \multicolumn{2}{c}{\textbf{AA}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{AB}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{BB}} \\

        \multicolumn{1}{c}{\textbf{DO-361}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-361}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-361}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} \\
        \cline{2-3}\cline{6-7}\cline{10-11}
      \textbf{AA} & 99.7\% &  0.3\% &&
      \textbf{AA} & 67.6\% & 32.4\% &&
      \textbf{AA} & 34.7\% & 65.3\% \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{AB} & 84.0\%   & 16.0\%  &&
      \textbf{AB} & 51.3\%   & 48.7\%  &&
      \textbf{AB} & 17.1\%   & 82.9\%  \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{BB} & 67.7\% & 32.3\% &&
      \textbf{BB} & 34.3\% & 65.7\% &&
      \textbf{BB} &  0.6\% & 99.4\% \\
        \cline{2-3}\cline{6-7}\cline{10-11}

    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\color{white} 1,306,406}
                & \multicolumn{1}{c}{\color{white} 75,326} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\color{white} 339,397}
                & \multicolumn{1}{c}{\color{white} 162,432} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\color{white} 33,661}
                & \multicolumn{1}{c}{\color{white} 93,162}
  \end{tabular} }
}


\note{
}

\end{frame}



\begin{frame}{Genotype pair vs microbiome reads}


\bigskip {\hilit genomic DO-361 and DO-362 vs microbiome DO-361}

\bigskip \bigskip

{\hspace*{-9mm}
  \footnotesize
  \renewcommand{\arraystretch}{2}
  \setlength{\tabcolsep}{1.5mm}
\only<1>{  \begin{tabular}{c|r|r|cc|r|r|cc|r|r|}
        \multicolumn{1}{c}{\textbf{DO-361:}} & \multicolumn{2}{c}{\textbf{AA}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{AB}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{BB}} \\

        \multicolumn{1}{c}{\textbf{DO-362}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-362}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-362}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} \\
        \cline{2-3}\cline{6-7}\cline{10-11}
      \textbf{AA} & 2,162,059 &   6,503 &&
      \textbf{AA} &   428,229 & 405,190 &&
      \textbf{AA} &     1,623 & 155,276 \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{AB} & 788,011 &   2,806 &&
      \textbf{AB} & 223,680 & 215,833 &&
      \textbf{AB} &   1,126 & 140,739 \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{BB} & 104,511 &     511 &&
      \textbf{BB} &  64,082 &  62,658 &&
      \textbf{BB} &   1,103 & 151,334 \\
        \cline{2-3}\cline{6-7}\cline{10-11}

    \multicolumn{1}{c}{} & \multicolumn{1}{c}{} & \multicolumn{1}{c}{} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{} & \multicolumn{1}{c}{} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{} & \multicolumn{1}{c}{}
  \end{tabular} }
\only<2|handout 0>{  \begin{tabular}{c|r|r|cc|r|r|cc|r|r|}
        \multicolumn{1}{c}{\textbf{DO-361:}} & \multicolumn{2}{c}{\textbf{AA}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{AB}} &&
        \multicolumn{1}{c}{} & \multicolumn{2}{c}{\textbf{BB}} \\

        \multicolumn{1}{c}{\textbf{DO-362}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-362}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} &&
        \multicolumn{1}{c}{\textbf{DO-362}}&
                 \multicolumn{1}{c}{\textbf{A}}&\multicolumn{1}{c}{\textbf{B}} \\
        \cline{2-3}\cline{6-7}\cline{10-11}
      \textbf{AA} & 99.7\% &  0.3\% &&
      \textbf{AA} & 51.4\% & 48.6\%  &&
      \textbf{AA} &  1.0\% & 99.0\% \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{AB} & 99.6\% &  0.4\% &&
      \textbf{AB} & 50.9\% & 49.1\%  &&
      \textbf{AB} &  0.8\% & 99.2\%  \\
        \cline{2-3}\cline{6-7}\cline{10-11}

      \textbf{BB} & 99.5\% &  0.5\% &&
      \textbf{BB} & 50.6\% & 49.4\% &&
      \textbf{BB} &  0.7\% & 99.3\% \\
        \cline{2-3}\cline{6-7}\cline{10-11}

    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\color{white} 2,162,059}
                & \multicolumn{1}{c}{\color{white} 6,503} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\color{white} 428,229}
                & \multicolumn{1}{c}{\color{white} 405,190} &&
    \multicolumn{1}{c}{} & \multicolumn{1}{c}{\color{white} 1,623}
                & \multicolumn{1}{c}{\color{white} 155,276}
  \end{tabular} }
}

\note{
}

\end{frame}



\begin{frame}[c]{log likelihood vs contaminant proportion}

\figw{Figs/lrt_v_contam.pdf}{1.0}

\note{
}

\end{frame}




\begin{frame}[c]{Detailed results}

\figw{Figs/lrt_v_contam_detail.pdf}{1.0}

\note{
}

\end{frame}



\begin{frame}[c]{Is everything contaminated?}

\figw{Figs/lrt_v_contam_expand.pdf}{1.0}

\note{
}

\end{frame}




\begin{frame}[c]{Summary}

\bbi
 \item Microbiome shotgun reads include reads from the host
 \item With such data, sample mix-ups can be identified
 \item Simple method:
   \bi
 \item Impute genotype at all SNPs
 \item Count alleles in reads overlapping SNPs
 \item Focus on homozygous SNPs and calculate percent discordant reads
   \ei
 \item We also saw strong evidence for many samples being mixtures
\ei

\note{
}

\end{frame}




\begin{frame}[c]{Acknowledgments}

\bbi
 \item Lindsay Traeger
 \item Alexandra Lobo
 \item Federico Rey
 \item Alan Attie, Mark Keller, Gary Churchill, Brian Yandell
\ei

\note{
  Lindsay had the idea to look for these mix-ups. (She was a postdoc
  with Federico Rey in Microbiology at UW-Madison.) Alexandra did most
  of the work. (She was a summer student with me and now is a graduate
  student in the Biomedical Data Science PhD program at UW-Madison.

  This is part of a larger project looking at the genetics of
  diabetes, obesity, and related traits.
}

\end{frame}






\begin{frame}[c]{}

\large

Slides: \href{https://bit.ly/2019CTC}{\tt bit.ly/2019CTC} \quad
\includegraphics[height=5mm]{Figs/cc-zero.png}

\vspace{8mm}

bioRxiv manuscript: \href{https://doi.org/10.1101/529040}{{\tt doi.org/10.1101/529040}}

\vspace{8mm}

\href{https://kbroman.org}{\tt kbroman.org}

\vspace{8mm}

\href{https://github.com/kbroman}{\tt github.com/kbroman}

\vspace{8mm}

\href{https://twitter.com/kwbroman}{\tt @kwbroman}


\note{
  Here's where you can find me, as well as the slides for this talk.

  Also note that there is a bioRxiv manuscript describing the details
  of this work.
}
\end{frame}




\end{document}
